include/openssl/core_names.h

```diff
@@ -240,6 +241,7 @@ extern "C" {
 #define OSSL_PKEY_PARAM_MGF1_PROPERTIES     "mgf1-properties"
 #define OSSL_PKEY_PARAM_TLS_ENCODED_PT      "tls-encoded-pt"
 #define OSSL_PKEY_PARAM_GROUP_NAME          "group"
+#define OSSL_PKEY_PARAM_DIST_ID             "distinguished-id"
```

Comment:
For legacy, the name is "distid"...  you need to either translate that to this in `legacy_ctrl_str_to_param()` (`crypto/evp/pmeth_lib.c`), or change this to "distid"...

If it was my choice, I would do the latter, i.e. keep the old name.  I leave it to youor preference, though...

Reply:
Actually, because of "hexdistid", it's easier to give this macro the value "distid"